cvs diff: Diffing basctl
cvs diff: Diffing basctl/inc
cvs diff: Diffing basctl/inc/pch
cvs diff: Diffing basctl/prj
cvs diff: Diffing basctl/sdi
cvs diff: Diffing basctl/source
cvs diff: Diffing basctl/source/accessibility
cvs diff: Diffing basctl/source/basicide
Index: basctl/source/basicide/basides1.cxx
===================================================================
RCS file: /cvs/script/basctl/source/basicide/basides1.cxx,v
retrieving revision 1.53
retrieving revision 1.53.24.1
diff -u -p -b -w -B -r1.53 -r1.53.24.1
--- basctl/source/basicide/basides1.cxx	9 Oct 2007 15:21:36 -0000	1.53
+++ basctl/source/basicide/basides1.cxx	8 Jan 2008 14:31:04 -0000	1.53.24.1
@@ -1146,7 +1146,8 @@ void BasicIDEShell::SetCurWindow( IDEBas
 		if ( pCurWin )
 		{
 			SetWindow( pCurWin );
-            SfxObjectShell::SetWorkingDocument( pCurWin->GetDocument().getDocumentOrNull() );
+            if ( pCurWin->GetDocument().isDocument() )
+                SfxObjectShell::SetWorkingDocument( pCurWin->GetDocument().getDocument() );
 		}
 		else
 		{
@@ -1365,8 +1366,8 @@ void __EXPORT BasicIDEShell::Activate( B
 {
 	if ( bMDI )
 	{
-        if ( pCurWin )
-            SfxObjectShell::SetWorkingDocument( pCurWin->GetDocument().getDocumentOrNull() );
+        if ( pCurWin && pCurWin->GetDocument().isDocument() )
+            SfxObjectShell::SetWorkingDocument( pCurWin->GetDocument().getDocument() );
 
 		if( pCurWin && pCurWin->IsA( TYPE( DialogWindow ) ) )
 			((DialogWindow*)pCurWin)->UpdateBrowser();
@@ -1418,10 +1419,6 @@ void __EXPORT BasicIDEShell::Deactivate(
 */
 
 		ShowObjectDialog( FALSE, FALSE );
-
-        Reference< XModel > xWorkingDoc( SfxObjectShell::GetWorkingDocument() );
-        if ( pCurWin && xWorkingDoc.is() && pCurWin->IsDocument( ScriptDocument( xWorkingDoc ) ) )
-            SfxObjectShell::SetWorkingDocument( NULL );
 	}
 }
 
Index: basctl/source/basicide/basidesh.cxx
===================================================================
RCS file: /cvs/script/basctl/source/basicide/basidesh.cxx,v
retrieving revision 1.45
retrieving revision 1.45.24.1
diff -u -p -b -w -B -r1.45 -r1.45.24.1
--- basctl/source/basicide/basidesh.cxx	9 Oct 2007 15:22:02 -0000	1.45
+++ basctl/source/basicide/basidesh.cxx	8 Jan 2008 14:31:05 -0000	1.45.24.1
@@ -242,7 +242,8 @@ __EXPORT BasicIDEShell::~BasicIDEShell()
 	SetWindow( 0 );
 	SetCurWindow( 0 );
     SfxObjectShell* pCurrentShell( SfxObjectShell::Current() );
-    SfxObjectShell::SetWorkingDocument( pCurrentShell ? pCurrentShell->GetModel() : Reference< frame::XModel >() );
+    if ( pCurrentShell )
+        SfxObjectShell::SetWorkingDocument( pCurrentShell->GetModel() );
 
 	// Alle Fenster zerstoeren:
 	IDEBaseWindow* pWin = aIDEWindowTable.First();
cvs diff: Diffing basctl/source/dlged
cvs diff: Diffing basctl/source/inc
cvs diff: Diffing basctl/uiconfig
cvs diff: Diffing basctl/uiconfig/basicide
cvs diff: Diffing basctl/uiconfig/basicide/accelerator
cvs diff: Diffing basctl/uiconfig/basicide/accelerator/en-GB
cvs diff: Diffing basctl/uiconfig/basicide/accelerator/en-US
cvs diff: Diffing basctl/uiconfig/basicide/menubar
cvs diff: Diffing basctl/uiconfig/basicide/statusbar
cvs diff: Diffing basctl/uiconfig/basicide/toolbar
cvs diff: Diffing basctl/util
